app/tests/crud/test_test_run_execution.py

```diff
@@ -343,6 +344,77 @@ def test_create_test_run_execution_from_test_run_config(db: Session) -> None:
         assert missing_count == 0
 
 
+def test_create_test_run_execution_certification_from_selected_tests_mandatory_suite(
+    db: Session,
+) -> None:
+    first_test_suite_identifier = "SampleTestSuite1"
+    first_test_case_identifier = "TCSS1001"
+    first_mandatory_test_case_identifier = "TCSS3001"
+
+    selected_tests = {
+        "sample_tests": {
+            first_test_suite_identifier: {
+                first_test_case_identifier: 1,
+                "TCSS1002": 2,
+                "TCSS1003": 3,
+            },
+            "SampleTestSuite2": {
+                "TCSS2001": 2,
+            },
+            "SampleTestSuite3Mandatory": {
+                "TCSS3001": 1,
+            },
+        },
+    }
+
+    total_test_case_count = (
+        sum(selected_tests["sample_tests"][first_test_suite_identifier].values())
+        + sum(selected_tests["sample_tests"]["SampleTestSuite2"].values())
+        + sum(selected_tests["sample_tests"]["SampleTestSuite3Mandatory"].values())
+    )
+
+    # Prepare data for test_run_execution
+    test_run_execution_title = "Test Execution title"
+    test_run_execution_data = TestRunExecutionCreate(
+        title=test_run_execution_title, certification_mode=True
+    )
+
+    test_run_execution = crud.test_run_execution.create(
+        db=db, obj_in=test_run_execution_data, selected_tests=selected_tests
+    )
+
+    # Assert direct properties
+    assert test_run_execution.title == test_run_execution_title
+    assert test_run_execution.test_run_config_id is None
+    assert test_run_execution.certification_mode
+
+    # Assert created test_suite_executions
+    test_suite_executions = test_run_execution.test_suite_executions
+    assert len(test_suite_executions) > 0
+
+    total_test_case_executions = 0
+    for test_suite_execution in test_suite_executions:
+        total_test_case_executions += len(test_suite_execution.test_case_executions)
+
+    assert total_test_case_executions == total_test_case_count
+
+    first_test_case_execution = test_suite_executions[0].test_case_executions[0]
+    assert first_test_case_execution.public_id == first_mandatory_test_case_identifier
+
+    remaining_test_cases = selected_tests["sample_tests"]
```

Comment:
```suggestion
    remaining_test_suites = selected_tests["sample_tests"]
```

Reply:
The goal of this `selected_tests["sample_tests"]`, is to retrieve all test cases inside the all suites.